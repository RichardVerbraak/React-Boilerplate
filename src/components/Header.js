import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

// NavLink is just a hyperlink to another page
export const Header = (props) => (
    <header className="header">
    <div className="content-container">
        <div className="header__content">         
            <Link className="header__title" to="/dashboard">
                <h1>Boilerplate</h1>
            </Link>
            <button className="button button--link" onClick={props.startLogout}>Logout</button>
        </div>
    </div>   
    </header>
)

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}

export default connect(undefined, mapDispatchToProps)(Header)