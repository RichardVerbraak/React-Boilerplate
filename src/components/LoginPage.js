import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'
 
export const LoginPage = (props) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Boilerplate</h1>
                <p>Tag line for app</p>
                <button className="button" onClick={props.startLogin}>Login with Google</button>
            </div>        
        </div>
    )
}

// Give the props access to the dispatch function
const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () => dispatch(startLogin())
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage)