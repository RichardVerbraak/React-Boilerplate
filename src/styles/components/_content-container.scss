// to distribute the left over space from max-width, we use margin 0 auto to have equal margins on left and right
.content-container {
    margin: 0 auto;
    padding: 0 $m-size;
    max-width: 80rem;
}