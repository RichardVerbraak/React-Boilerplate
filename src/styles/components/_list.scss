.list-header {
    background: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-body {
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}

// transition targets the background and eases in the off-white
// &:hover is saying, also give a hover effect to list-item
.list-item {
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }
    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $grey;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__subtitle {
    color: $grey;
    font-size: $font-size-small;
}

.list-item__amount {
    margin: $s-size 0 0 0;

    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}