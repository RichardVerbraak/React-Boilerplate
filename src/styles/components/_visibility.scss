// displays expenses only on mobile
// If it reaches min desktop width -> dont show
.show-for-mobile {
    @media (min-width: $desktop-breakpoint) {
        display: none;
    }
}

// displays expense / amount on desktop
.show-for-desktop {
    @media (max-width: $desktop-breakpoint - 0.1rem) {
        display: none;
    }
}