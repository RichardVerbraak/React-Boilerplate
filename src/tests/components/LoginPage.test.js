import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../../components/LoginPage'

let wrapper, startLogin

beforeEach(() => {
    startLogin = jest.fn()
    wrapper = shallow(<LoginPage startLogin={startLogin}></LoginPage>)
})

test('Should render LoginPage', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should call startLogout on button click', () => {    
    wrapper.find('button').simulate('click')
    expect(startLogin).toHaveBeenCalled()
})