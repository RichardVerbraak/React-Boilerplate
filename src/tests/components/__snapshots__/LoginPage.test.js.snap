// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      Tag line for app
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
